### PR TITLE
Adds filters to customize product title classes, control product exce…

### PR DESCRIPTION
**Pull request by hsankala**
OK third time lucky creating this PR! 

Disussion about the PR here: https://github.com/orgs/bootscore/discussions/883 

### Summary:  
Enhancements to the Ajax Mini Cart: product excerpt display, stock level notifications, and dynamic product title classes.  

#### Product Excerpt Display  
- A new filter has been added (enabled by default).  
- When disabled, the product excerpt will not be displayed in the Ajax Mini Cart.  

#### Stock Level Notifications  
- Added a filter (on by default) to toggle stock level notifications.  
- Disabling this filter removes stock notifications from the Ajax Mini Cart.  

#### Product Titles  
- Introduced a filter for programmatic access to CSS classes on product titles.  
- Added a more specific title class (cart-product-title) to allow developers to add custom classes easily.  

Testing
____________

To test this, I added two filter in functions.php that disables the enable cart product excerpt feature. I also added a filter targeting the enable cart stock quantity feature and disabled that as well. Additionally, I used the product title filter to remove the truncated text class, effectively eliminating the specific truncation CSS class. Finally, I examined the source code for the specific title class for the cart-product-title and confirmed that it was present.

```
/**
 * Disable cart_product_excerp
 */
add_filter('bootscore/class/cart/enable_cart_product_excerpt', '__return_false');

/**
 * Disable cart_stock_quantity
 */
add_filter('bootscore/class/cart/enable_cart_stock_quantity', '__return_false');


function remove_text_truncate_class($classes) {
    // Remove the 'text-truncate' class if it exists
    $classes = str_replace('text-truncate', '', $classes);
    
    // Trim any leading or trailing spaces that might be left
    $classes = trim($classes);
    
    // Return the modified class list
    return $classes;
}
add_filter('bootscore/class/cart/product-title', 'remove_text_truncate_class');
```

### CODE REVIEW

**Comment by crftwrk** [comment]
Hi Hefin,

thank you, this PR is in the correct branch now. I have just two tiny suggestions (comments). If you agree with requested changes, it would be great if you just change them.

**Comment by hsankala** [comment]
Yo Basti! I’ve checked the "Files changed" tab under "Review changes", but I can’t find any comments. I’m happy to make any requested changes—just need a little guidance on where to find them.

Best regards,
Hefin

**Comment by crftwrk** [comment]
Hi Hefin,

just scroll up a bit, there you will find two suggested  changes ;-)

Or take a look to the files changed https://github.com/bootscore/bootscore/pull/887/files

**Comment by hsankala** [comment]
Forgive me if I'm being denser than a neutron star. But I don't see any updates on the '[Files changed tab](https://github.com/bootscore/bootscore/pull/887/files)'.  These were the orignial changes

image

And scrolling up on this PR convo doesn't mention any changes to make.

**Comment by crftwrk** [comment]
Sorry, I forgot to submit the review. Now they are there.

File: woocommerce/cart/mini-cart.php [hunk 1]
```diff
@@ -62,27 +62,32 @@
...
-              <small class="text-body-secondary d-block text-truncate">
-                <?php echo get_the_excerpt($product_id); ?>
-              </small>
+              <?php
+                if (apply_filters('bootscore/class/cart/enable_cart_product_excerpt', true)) { ?>  
+                  <small class="text-body-secondary d-block text-truncate">
+                    <?php echo get_the_excerpt($product_id); ?>
+                  </small>
+              <?php  }
+              ?>
```
[the remark this answers is not in the document]
Comment (hsankala) [reply woocommerce/cart/mini-cart.php:77]: Yes! Makes sense to have adjustability here too. Tweaks!

File: woocommerce/cart/mini-cart.php [hunk 2]
```diff
@@ -62,27 +62,32 @@
...
-                // Check if the product is sold individually
-                if ($_product->is_sold_individually()) {
-                  echo '<div class="cart-badge mb-2"><span class="badge bg-danger">' . esc_html__('Sold individually', 'woocommerce') . '</span></div>';
-                }
-
-                // Check if the product has only 5 or fewer left in stock
-                elseif ($stock_quantity <= 5 && $stock_quantity > 0) {
-                  $stock_message = sprintf(esc_html__('Only %s left in stock', 'woocommerce'), $stock_quantity);
-                  echo '<div class="cart-badge mb-2"><span class="badge bg-danger">' . $stock_message . '</span></div>';
+                if (apply_filters('bootscore/class/cart/enable_cart_stock_quantity', false)) {
```
[the remark this answers is not in the document]
Comment (hsankala) [reply woocommerce/cart/mini-cart.php:80]: Agreed.  Better show in the first place so they can decided if they want to hide it.

**Comment by hsankala** [comment]
OK actioned the review points above. Should be good to go!

**Review by crftwrk — APPROVED** [review]
This is perfect, thank you! If you have more ideas about filters or actions, feel free to create a PR to `Add more filters` branch anytime.

**Pull request merged**